hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/placement/algorithms/SCMContainerPlacementRackScatter.java

```diff
@@ -98,6 +98,7 @@ public List<DatanodeDetails> chooseDatanodes(
       throws SCMException {
     Preconditions.checkArgument(nodesRequired > 0);
     metrics.incrDatanodeRequestCount(nodesRequired);
```

Comment:
I think the changes looks good to me.
To keep the better practice: How about we make all the passed params final. And have a separate counter before while loop?

int nodesRequiredToChoose = nodesRequired;
while (nodesRequiredToChoose > 0) {

Reply:
sure, will do this.